transaction-pool/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+use solana_sdk::pubkey::Pubkey;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::collections::VecDeque;
+
+#[derive(Default, Clone, PartialEq, Debug)]
+pub struct Item {
+    pub priority: u64,
+    pub units: u64,
+    pub time: u64,
+    pub id: (u32, u32),
+}
+
+#[derive(Default, Clone)]
+pub struct Pool {
+    pub max_age: u64,
+    pub max_bucket_cu: u64,
+    pub max_block_cu: u64,
+    pub count: usize,
+    items: BTreeMap<u64, VecDeque<Item>>,
+}
+
+pub trait Table {
+    fn keys(&self, item: &Item) -> &[&Pubkey];
+}
+
+impl Pool {
+    pub fn insert(&mut self, item: Item) {
+        let bucket = self
+            .items
+            .entry(item.priority)
+            .or_insert_with(VecDeque::new);
+        bucket.push_back(item);
+        self.count += 1;
```

Comment:
Should `pool` has max capacity, say up to 128,000 packets; Drop low priority packets when `self.count > self.capacity` by `self.first_entry().pop_front()`? When a bucket is empty after pop, maybe OK to keep the empty bucket in tree.